src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/RazorLSPTextDocumentCreatedListener.cs

```diff
@@ -161,6 +183,29 @@ private void InitializeRazorLSPTextBuffer(ITextBuffer textBuffer)
                 // Must track the document after changing the content type so any LSPDocuments created understand they're being created for a Razor LSP document.
                 _lspDocumentManager.TrackDocument(textBuffer);
             }
+
+            // Initialize the buffer with editor options.
+            InitializeOptions(textBuffer);
```

Comment:
We can't rely on ourselves being on a RemoteClient so lets move this piece into the non-remote client path prior to the `ChangeContentType` so anyone listening for those events gets the proper data.